lnd_test.go

```diff
@@ -6812,14 +6818,29 @@ func testSwitchCircuitPersistence(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("Node restart failed: %v", err)
 	}
 
+	// Ensure all nodes in the network still have 5 outstanding htlcs.
+	err = lntest.WaitPredicate(func() bool {
+		return assertNumActiveHtlcs(nodes, numPayments)
+	}, time.Second*15)
+	if err != nil {
+		t.Fatalf("htlc mismatch: %v", err)
+	}
+
 	// Now restart carol without hodl mode, to settle back the outstanding
 	// payments.
 	carol.SetExtraArgs(nil)
 	if err := net.RestartNode(carol, nil); err != nil {
 		t.Fatalf("Node restart failed: %v", err)
 	}
 
-	time.Sleep(time.Second * 5)
+	// After the payments settle, there should be no active htlcs on any of
+	// the nodes in the network.
+	err = lntest.WaitPredicate(func() bool {
+		return assertNumActiveHtlcs(nodes, 0)
+	}, time.Second*15)
+	if err != nil {
+		t.Fatalf("htlc mismatch: %v", err)
```

Comment:
Looks like this still failing on Travis :/

Reply:
I think I have a fix, from the logs it didn't look like then nodes were being reconnected. I've added an `EnsureConnected` method which is like `ConnectNodes`, but doesn't fail if we are connected or have a pending connection request. Looking good so far :)